src/components/molecules/CameraMicrophoneControls/CameraMicrophoneControls.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useCallback } from "react";
+import classNames from "classnames";
+import Video, {
+  AudioTrackPublication,
+  VideoTrackPublication,
+} from "twilio-video";
+
+import { ContainerClassName } from "types/utility";
+
+import { VideoOverlayButton } from "components/atoms/VideoOverlayButton";
+
+import "components/molecules/CameraMicrophoneControls/CameraMicrophoneControls.scss";
+
+export interface CameraMicrophoneControlsProps extends ContainerClassName {
+  participant: Video.Participant;
+  defaultMute: boolean;
+}
+
+export const CameraMicrophoneControls: React.FC<CameraMicrophoneControlsProps> = ({
+  participant,
+  defaultMute,
+  containerClassName,
+}) => {
+  const getChangeStateHandler = useCallback(
+    (variant: "microphone" | "camera"): ((enabled: boolean) => void) => {
+      const tracks =
+        variant === "microphone"
+          ? participant.audioTracks
+          : participant.videoTracks;
+
+      return (enable: boolean) => {
+        tracks.forEach(
+          (track: AudioTrackPublication | VideoTrackPublication) => {
+            const innerTrack = track.track;
+            if (innerTrack) {
+              if (enable && "enable" in innerTrack) innerTrack.enable();
+              else if ("disable" in innerTrack) innerTrack.disable();
+            }
+          }
+        );
+      };
+    },
+    [participant.audioTracks, participant.videoTracks]
+  );
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const changeAudioState = useCallback(getChangeStateHandler("microphone"), [
+    getChangeStateHandler,
+  ]);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const changeVideoState = useCallback(getChangeStateHandler("camera"), [
+    getChangeStateHandler,
+  ]);
```

Comment:
Why not exhaustive deps?

`getChangeStateHandler` doesn't have to return a function. `useCallback()` already solves the issue of creating multiple functions.

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/2221/commits/b1584560cb5a102ef1a5bf47411f29275f37b717